katharsis-rs/src/main/java/io/katharsis/resource/registry/UriInfoServiceUrlProvider.java

```diff
@@ -2,17 +2,44 @@
 
 import javax.ws.rs.core.UriInfo;
 
-
 public class UriInfoServiceUrlProvider implements ServiceUrlProvider {
 
-    private UriInfo info;
+	@Deprecated
+	private UriInfo globalUrlInfo;
+
+	private ThreadLocal<UriInfo> threadLocal = new ThreadLocal<>();
+
+	public UriInfoServiceUrlProvider() {
+		// make use of thread local and started/finish events to get uriInfo
+	}
+
+	@Deprecated
+	public UriInfoServiceUrlProvider(UriInfo info) {
+		this.globalUrlInfo = info;
+	}
+
+	@Override
+	public String getUrl() {
+		UriInfo urlInfo = globalUrlInfo;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "globalUrlInfo"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)